File: Algorithms/Game Theory/Introduction to Nim Game.php
<?php

/*
 * Complete the 'nimGame' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER_ARRAY pile as parameter.
 */

function nimGame($pile) {
    # Write your code here
    $xor = 0;
    foreach($pile as $p){
        $xor ^= $p;
    }
    return $xor == 0 ? "Second" : "First";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$g = intval(trim(fgets(STDIN)));

for ($g_itr = 0; $g_itr < $g; $g_itr++) {
    $n = intval(trim(fgets(STDIN)));

    $pile_temp = rtrim(fgets(STDIN));

    $pile = array_map('intval', preg_split('/ /', $pile_temp, -1, PREG_SPLIT_NO_EMPTY));

    $result = nimGame($pile);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);

File: Algorithms/Game Theory/Introduction to Nim Game.py
import os

#
# Complete the 'nimGame' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER_ARRAY pile as parameter.
#

def nimGame(pile):
    # Write your code here
    xor_sum = 0
    for p in pile:
        xor_sum ^= p

    if xor_sum == 0:
        return "Second"
    else:
        return "First"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    g = int(input().strip())

    for g_itr in range(g):
        n = int(input().strip())

        pile = list(map(int, input().rstrip().split()))

        result = nimGame(pile)

        fptr.write(result + '\n')

    fptr.close()
